server/routes/stats_router.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ResponseError } from '@opensearch-project/opensearch/lib/errors';
+import {
+  IOpenSearchDashboardsResponse,
+  IRouter,
+  SavedObjectsFindResponse,
+} from '../../../../src/core/server';
+import { APP_API, APP_PATH, MAP_SAVED_OBJECT_TYPE } from '../../common';
+import { getStats } from '../common/stats/stats_helper';
+import { MapSavedObjectAttributes } from '../../common/map_saved_object_attributes';
+
+export function statsRoute(router: IRouter) {
+  router.get(
+    {
+      path: `${APP_API}${APP_PATH.STATS}`,
+      validate: false,
+    },
+    async (
+      context,
+      request,
+      response
+    ): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
+      try {
+        const savedObjectsClient = context.core.savedObjects.client;
+        const mapsSavedObjects: SavedObjectsFindResponse<MapSavedObjectAttributes> =
+          await savedObjectsClient?.find({ type: MAP_SAVED_OBJECT_TYPE });
+        const stats = getStats(mapsSavedObjects);
```

Comment:
what will happen if mapsSavedObjects == null? how we are handling that case?

Reply:
mapsSavedObjects is come from savedObjectsClient?.find(), if the map type not find, mapsSavedObjects will be 
```
{
  page: 1,
  per_page: 1000,
  total: 0,
  saved_objects: [
    {
    
    }
  ]
}
```
and the API return will be:

```
{
"maps_total": 0,
"layers_filters_total": 0,
"layers_total": {
"opensearch_vector_tile_map": 0,
"documents": 0,
"wms": 0,
"tms": 0
},
"maps_list": []
}
```

Reply:
thanks. I think I missed reading that "?" in the code.